Cubical/ZCohomology/CohomologyRings/S2wedgeS4.agda

```diff
@@ -0,0 +1,522 @@
+{-# OPTIONS --safe --experimental-lossy-unification #-}
+
+module Cubical.ZCohomology.CohomologyRings.S2wedgeS4 where
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Function
+open import Cubical.Foundations.Isomorphism
+open import Cubical.Foundations.Transport
+open import Cubical.Foundations.HLevels
+
+open import Cubical.Relation.Nullary
+
+open import Cubical.Data.Empty as ⊥
+open import Cubical.Data.Unit
+open import Cubical.Data.Bool
+open import Cubical.Data.Nat using (ℕ ; zero ; suc ; discreteℕ ; suc-predℕ ; +-comm)
+open import Cubical.Data.Int
+open import Cubical.Data.Sigma
+open import Cubical.Data.Sum
+open import Cubical.Data.Vec
+open import Cubical.Data.FinData
+
+open import Cubical.Algebra.Group
+open import Cubical.Algebra.Group.Morphisms
+open import Cubical.Algebra.Group.MorphismProperties
+open import Cubical.Algebra.AbGroup
+open import Cubical.Algebra.Group.Instances.Unit
+open import Cubical.Algebra.Group.Instances.Bool
+open import Cubical.Algebra.Group.Instances.Int
+open import Cubical.Algebra.DirectSum.DirectSumHIT.Base
+open import Cubical.Algebra.Ring
+
+open import Cubical.Algebra.CommRing
+open import Cubical.Algebra.CommRing.FGIdeal
+open import Cubical.Algebra.CommRing.QuotientRing
+open import Cubical.Algebra.CommRing.Instances.Int renaming (ℤCommRing to ℤCR)
+open import Cubical.Algebra.CommRing.Instances.Polynomials.MultivariatePoly
+open import Cubical.Algebra.CommRing.Instances.Polynomials.MultivariatePoly-Quotient
+open import Cubical.Algebra.CommRing.Instances.Polynomials.MultivariatePoly-notationZ
+
+open import Cubical.HITs.Truncation
+open import Cubical.HITs.SetTruncation as ST
+open import Cubical.HITs.PropositionalTruncation as PT
+open import Cubical.HITs.SetQuotients as SQ renaming (_/_ to _/sq_)
+open import Cubical.HITs.RPn.Base
+
+open import Cubical.ZCohomology.Base
+open import Cubical.ZCohomology.GroupStructure
+open import Cubical.ZCohomology.RingStructure.CupProduct
+open import Cubical.ZCohomology.RingStructure.RingLaws
+open import Cubical.ZCohomology.RingStructure.CohomologyRing
+open import Cubical.ZCohomology.Groups.S2wedgeS4
+open import Cubical.ZCohomology.CohomologyRings.CupProductProperties
+
+open Iso
+
+
+module Equiv-RP2-Properties where
+
+-----------------------------------------------------------------------------
+-- Definitions, Import with notations, Partition
+
+  -- Definition
+  private
+    ℤAG = Ring→AbGroup (CommRing→Ring ℤCR)
+
+  <XY,X²,Y²> : FinVec ℤ[x,y] 3
+  <XY,X²,Y²> zero = base (1 ∷ 1 ∷ []) 1
+  <XY,X²,Y²> one = base (2 ∷ 0 ∷ []) 1
+  <XY,X²,Y²> two = base (0 ∷ 2 ∷ []) 1
+
+  ℤ[X,Y]/<XY,X²,Y²> : CommRing ℓ-zero
+  ℤ[X,Y]/<XY,X²,Y²> = PolyCommRing-Quotient ℤCR <XY,X²,Y²>
+
+  ℤ[x,y]/<xy,x²,y²> : Type ℓ-zero
+  ℤ[x,y]/<xy,x²,y²> = fst ℤ[X,Y]/<XY,X²,Y²>
+
+  -- Import with notation
+  open IsGroupHom
+  open gradedRingProperties
+
+  open CommRingStr (snd ℤCR) using ()
```

Comment:
Didn't we conclude that there was a nicer way to do these big renamings?

Reply:
I don't remember, overall it is a bit long but I think it is clearer when reading the file